app/controllers/account/pages_controller.rb

```diff
@@ -39,6 +39,17 @@ def create
     end
   end
 
+  def destroy
+    @page = resource
```

Comment:
is it necessary to always execute code at this line? can you initialize object after checking user`s permissions?